game-core/src/main/java/games/strategy/triplea/ui/menubar/TripleAMenuBar.java

```diff
@@ -60,60 +56,19 @@ private void createLobbyMenu(final InGameLobbyWatcherWrapper watcher) {
    *     cancelled the operation.
    */
   public static File getSaveGameLocation(final Frame frame) {
-    // For some strange reason, the only way to get a Mac OS X native-style file dialog
-    // is to use an AWT FileDialog instead of a Swing JDialog
-    if (SystemProperties.isMac()) {
-      final FileDialog fileDialog = new FileDialog(frame);
-      fileDialog.setMode(FileDialog.SAVE);
-      fileDialog.setDirectory(ClientSetting.saveGamesFolderPath.getValueOrThrow().toString());
-      fileDialog.setFilenameFilter((dir, name) -> GameDataFileUtils.isCandidateFileName(name));
-      fileDialog.setVisible(true);
+    final FileDialog fileDialog = new FileDialog(frame);
+    fileDialog.setMode(FileDialog.SAVE);
+    fileDialog.setDirectory(ClientSetting.saveGamesFolderPath.getValueOrThrow().toString());
+    fileDialog.setFilenameFilter((dir, name) -> GameDataFileUtils.isCandidateFileName(name));
+    fileDialog.setVisible(true);
 
-      final String fileName = fileDialog.getFile();
-      if (fileName == null) {
-        return null;
-      }
-
-      // If the user selects a filename that already exists,
-      // the AWT Dialog on Mac OS X will ask the user for confirmation
-      return new File(fileDialog.getDirectory(), GameDataFileUtils.addExtensionIfAbsent(fileName));
-    }
-
-    // Non-Mac platforms should use the normal Swing JFileChooser
-    final JFileChooser fileChooser = SaveGameFileChooser.getInstance();
-    final int selectedOption = fileChooser.showSaveDialog(frame);
-    if (selectedOption != JFileChooser.APPROVE_OPTION) {
+    final String fileName = fileDialog.getFile();
+    if (fileName == null) {
       return null;
     }
-    File f = fileChooser.getSelectedFile();
-    // disallow sub directories to be entered (in the form directory/name) for Windows boxes
```

Comment:
Ok, this is a super weird thing I have never seen before.

Reply:
Always quite a bit of 'interesting' code to find here and there  :grin:

Reply:
I mean at this point I shouldn't really be surprised anymore, but it happens again and again ^^